Lib/fontParts/base/glyph.py

```diff
@@ -2152,70 +3100,104 @@ def removeLayer(self, layer):
+
+            Subclasses may override this method.
 
-        Subclasses may override this method.
         """
         self.raiseNotImplementedError()
 
     # -----
     # Image
     # -----
 
-    image = dynamicProperty(
+    image: dynamicProperty = dynamicProperty(
         "base_image",
-        "The :class:`BaseImage` for the glyph."
+        """Get the image for the glyph.
+
+        :return: The :class:`BaseImage` instance belonging to the glyph.
+
+        """
     )
 
-    def _get_base_image(self):
+    def _get_base_image(self) -> BaseImage:
         image = self._get_image()
         if image.glyph is None:
             image.glyph = self
         return image
 
-    def _get_image(self):
-        """
-        Subclasses must override this method.
+    def _get_image(self) -> BaseImage:
+        """Get the image for the native glyph.
+
+        :return: The :class:`BaseImage` subclass instance belonging to the glyph.
+        :raises NotImplementedError: If the method has not been overridden by a
+            subclass.
+
+        .. important::
+
+            Subclasses must override this method.
+
         """
         self.raiseNotImplementedError()
 
-    def addImage(self, path=None, data=None, scale=None,
-                 position=None, color=None):
-        """
-        Set the image in the glyph. This will return the
-        assigned :class:`BaseImage`. The image data can be
-        defined via ``path`` to an image file:
+    def addImage(self,
+        path: Optional[str] = None,
+        data: Optional[bytes] = None,
+        scale: ScaleType = None,
+                 position: CoordinateType = None,
+                 color: ColorType = None) -> BaseImage:
+        """Set the image in the glyph.
+
+        The image data may be provided as either the `path` to an image file or
+        directly as raw image `data`. The supported image formats will vary
+        across environments. Refer to :class:`BaseImage` for complete details.
+
+        :param path: The optional path to the image file to add to the glyph as
+            a :class:`str`. Defaults to :obj:`None`.
+        :param data: The optional raw image data to add to the glyph
+            as :class:`bytes`. Defaults to :obj:`None`.
+        :param scale: The optional x and y values to scale the glyph by as
+            a :class:`tuple` of :class:`int` or :class:`float` values.
+            Defaults to :obj:`None`.
+        :param position: The optional location of the lower left point of the
+            image as a :ref:`type-coordinate`. Defaults to :obj:`None`.
+        :param color: The optional color to be applied to the image as
+            a :ref:`type-color`. Defaults to :obj:`None`.
+        :return: The :class:`BaseImage` instance added to the glyph.
+        :raises IOError: If no valid image file can be found at the given path.
+        :raises FontPartsError:  If `path` and `data` are both provided.
+
+        Add the image as a filepath::
```

Comment:
No, I think we should only use `Example::` where there is no introductory text needed. Otherwise we should style them all as a heading with a custom directive (i.e.,  `:example:`) or strong formatting `**Example::**`.

Reply:
Agree, thank you!